api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/dataset/definition/SharedDataDefintion.java

```diff
@@ -36,4 +53,61 @@ public DataDefinition definition(String name, Concept concept) {
         obsForPersonDataDefinition.setWhich(TimeQualifier.LAST);
         return obsForPersonDataDefinition;
     }
+    public DataDefinition getPreARTNumber()
+    {
+        PatientIdentifierType preARTNo = MetadataUtils.existing(PatientIdentifierType.class, "e1731641-30ab-102d-86b0-7a5022ba4115");
+        DataDefinition identifierDefPre = new ConvertedPatientDataDefinition("identifier", new PatientIdentifierDataDefinition(preARTNo.getName(), preARTNo), identifierFormatter);
+        return identifierDefPre;
+    }
+    public PersonAttributeType getPatientNationality()
+    {
+        PersonAttributeType nationality = Context.getPersonService().getPersonAttributeTypeByUuid("dec484be-1c43-416a-9ad0-18bd9ef28929");
+         return nationality;
+    }
+    public PersonAttributeType getPhoneNumber()
+    {
+        PersonAttributeType phoneNumber = Context.getPersonService().getPersonAttributeTypeByUuid("14d4f066-15f5-102d-96e4-000c29c2a5d7");
+         return phoneNumber;
+    }
+    public PersonAttributeType getAlternatePhoneNumber()
+    {
+        PersonAttributeType alternatePhonenumber = Context.getPersonService().getPersonAttributeTypeByUuid("553834ef-b3fe-4c79-826a-6d4b6978bcff");
+        return alternatePhonenumber;
+    }
+    public  DataDefinition getNationalIDNumber()
+    {
+        PatientIdentifierType nationalIdentifiernumber = MetadataUtils.existing(PatientIdentifierType.class, "f0c16a6d-dc5f-4118-a803-616d0075d282");
+        DataDefinition nationalID = new ConvertedPatientDataDefinition("identifier", new PatientIdentifierDataDefinition(nationalIdentifiernumber.getName(), nationalIdentifiernumber), identifierFormatter);
+         return nationalID;
+    }
+
+    public Concept getConcept(String uuid) {
+        return Dictionary.getConcept(uuid);
+    }
+
+    public DataDefinition villageParish(){
+        CalculationDataDefinition cdf =new CalculationDataDefinition("village+parish", new PersonAddressCalculation());
+        cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+        return cdf;
+    }
+    public DataDefinition age(Integer lower, Integer upper) {
+        CalculationDataDefinition cdf = new CalculationDataDefinition("Age-"+lower+"-"+upper+"yrs", new AgeLimitCalculation());
+        cdf.addCalculationParameter("lowerLimit", lower);
+        cdf.addCalculationParameter("upperLimit", upper);
+        cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+        return cdf;
+    }
+    public DataDefinition whoCd4Vl(String q, String a){
```

Comment:
@solemabrothers rename to getWHOCD4ViralLoadCalculation (not even sure what this means) - what is it?

Reply:
This method returns the client CD4 and Viral Load

Reply:
Renamed